metricbeat/module/vsphere/datastorecluster/datastorecluster.go

```diff
@@ -55,6 +57,14 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	return &DatastoreClusterMetricSet{ms}, nil
 }
 
+type metricData struct {
+	assetNames assetNames
+}
+
+type assetNames struct {
+	outputDsNames []string
+}
+
 func (m *DatastoreClusterMetricSet) Fetch(ctx context.Context, reporter mb.ReporterV2) error {
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
```

Comment:
Since you are already passing a context, is it necessary to create a new one here?

Is it valuable to control its cancellation here?

Reply:
We have planned to remove this from all the metricset but not in current PR.
We will be having common PR for doing all the changes related to all the metricsets.